### PR TITLE
Added support for custom object serializer/deserializer

### PR DESCRIPTION
**Pull request by skeeet**
Particularly useful if you're unhappy with NSKeyedArchiver/unarchiver performance or want to encrypt persisted data.

### CODE REVIEW

**Comment by appleguy** [comment]
@skeeet ha, thanks a lot - we were actually just about to add this, had been discussing on Friday!  Really appreciate the contribution :).

**Comment by skeeet** [comment]
@appleguy You welcome. Actually had this for a while in private fork, but sharing is caring :)

**Comment by appleguy** [comment]
@garrettmoon any thoughts on this one?

File: PINCache/PINDiskCache.m [hunk 1]
```diff
@@ -116,6 +128,20 @@ - (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath
     return self;
 }
 
+- (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath serializer:(PINDiskCacheSerializerBlock)serializer deserializer:(PINDiskCacheDeserializerBlock)deserializer
```
Comment (garrettmoon) [inline PINCache/PINDiskCache.m]: Do you mind switching the implementation of this and `initWithName:(NSString *)name rootPath:` ? Then have initWithName:(NSString *)name rootPath call this one with the default serializer and deserializer?

File: PINCache/PINCache.m [hunk 2]
```diff
@@ -55,6 +55,23 @@ - (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath
     return self;
 }
 
+- (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath serializer:(PINDiskCacheSerializerBlock)serializer deserializer:(PINDiskCacheDeserializerBlock)deserializer
```
Comment (garrettmoon) [inline PINCache/PINCache.m]: Can you make `initWithName:rootPath:` call this method with nil serializer and deserializer?

**Comment by garrettmoon** [comment]
Thanks so much @skeeet !

**Pull request merged**